File: sdks/php/src/Types/StreamEvent.php
<?php

declare(strict_types=1);

namespace Synap\SDK\Types;

/**
 * Represents an event in a stream
 */
readonly class StreamEvent
{
    public function __construct(
        public int $offset,
        public string $event,
        public mixed $data,
        public int $timestamp = 0
    ) {
    }

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'offset' => $this->offset,
            'event' => $this->event,
            'data' => $this->data,
            'timestamp' => $this->timestamp,
        ];
    }
}

File: sdks/csharp/src/Synap.SDK/Types/StreamEvent.cs
using System.Text.Json;

namespace Synap.SDK.Types;

/// <summary>
/// Represents an event from a stream.
/// </summary>
public sealed record StreamEvent
{
    /// <summary>
    /// Gets the event offset in the stream.
    /// </summary>
    public required long Offset { get; init; }

    /// <summary>
    /// Gets the event type/name.
    /// </summary>
    public required string Event { get; init; }

    /// <summary>
    /// Gets the event data.
    /// </summary>
    public required JsonElement Data { get; init; }

    /// <summary>
    /// Gets the event timestamp (Unix timestamp in seconds).
    /// </summary>
    public long Timestamp { get; init; }

    /// <summary>
    /// Gets the room name.
    /// </summary>
    public string? Room { get; init; }
}

